BaseApp/COD57000.TXT

```diff
@@ -0,0 +1,40 @@
+OBJECT Codeunit 57000 DotNet Add Text Encoding
+{
+  OBJECT-PROPERTIES
+  {
+    Date=18-06-14;
+    Time=16:39:00;
+    Modified=Yes;
+    Version List=;
+  }
+  PROPERTIES
+  {
+    OnRun=BEGIN
+          END;
+
+  }
+  CODE
+  {
+    VAR
+      Encoding@1000 : DotNet "'mscorlib, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.Text.Encoding";
+      StreamWriter@1001 : DotNet "'mscorlib, Version=2.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'.System.IO.StreamWriter";
+
+    [External]
+    PROCEDURE AddTextLineWithEncoding@2(LineText@1000 : Text;VAR OutStream_@1001 : OutStream;CodePageNo@1002 : Integer);
```

Comment:
Hi,

Please put the Encoding dotnet variable in its own wrapper and add that encoding wrapper codeunit as the parameter here. The reason is that if someone else wants to initialize the StreamWriter with other types of encoding ) i.e. not using a CodePageNo then they can.


The code would look something like:


[External]
PROCEDURE StreamWriter@1(OutStream@1000 : OutStream;DotNet_Encoding@1001 : Codeunit 30xx);
VAR
  Encoding@1002 : DotNet "'mscorlib'.System.Text.Encoding";
BEGIN
  DotNet_Encoding.GetEncoding(Encoding);
  StreamWriter := StreamWriter.StreamWriter(OutStream,Encoding);
END;